### PR TITLE
Add support for configuring cluster virtual IP

### PR DESCRIPTION
**Pull request by 2ez4szliu**
An implementation for resource nsxt_cluster_virtual_ip to configure NSXT cluster virtual IPv4 and IPv6 address.

The virtual IP property can be part of nsxt_manager_cluster resource or an independent resource like in this PR, if we decide to implement it as part of nsxt_manage_cluster resource we can still use this PR as a reference.

### CODE REVIEW

File: nsxt/resource_nsxt_cluster_virtual_ip.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		Read:   resourceNsxtClusterVirualIPRead,
+		Update: resourceNsxtClusterVirualIPUpdate,
+		Delete: resourceNsxtClusterVirualIPDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"force": {
+				Type:         schema.TypeString,
```
Comment (annakhm) [inline nsxt/resource_nsxt_cluster_virtual_ip.go:36]: Any reason not to make this `TypeBool`?

File: nsxt/resource_nsxt_cluster_virtual_ip.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			"ip_address": {
+				Type:         schema.TypeString,
+				Description:  "Virtual IPv4 address",
+				Optional:     true,
+				ValidateFunc: validation.IsIPv4Address,
+				Default:      DefaultIPv4VirtualAddress,
+			},
+			"ipv6_address": {
+				Type:         schema.TypeString,
+				Description:  "IP address of the cluster node that will join the cluster of the host node",
```
Comment (annakhm) [inline nsxt/resource_nsxt_cluster_virtual_ip.go:51]: Is this description accurate?

File: nsxt/resource_nsxt_cluster_virtual_ip.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(ClusterVirtualIPForceType, false),
+				Default:      nsxModel.ClusterVirtualIpProperties_FORCE_FALSE,
+			},
+			"ip_address": {
+				Type:         schema.TypeString,
+				Description:  "Virtual IPv4 address",
+				Optional:     true,
+				ValidateFunc: validation.IsIPv4Address,
+				Default:      DefaultIPv4VirtualAddress,
```
Comment (annakhm) [inline nsxt/resource_nsxt_cluster_virtual_ip.go:47]: Are those defaults equivalent to "unspecified"?

File: nsxt/resource_nsxt_cluster_virtual_ip.go [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+	d.Set("ip_address", obj.IpAddress)
+	d.Set("ipv6_address", obj.Ip6Address)
+
+	return nil
+}
+
+func resourceNsxtClusterVirualIPUpdate(d *schema.ResourceData, m interface{}) error {
+
+	id := d.Id()
+	if id == "" {
```
Comment (annakhm) [inline nsxt/resource_nsxt_cluster_virtual_ip.go:86]: This is not needed in update flow

File: nsxt/resource_nsxt_cluster_virtual_ip.go [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+	connector := getPolicyConnector(m)
+	client := cluster.NewApiVirtualIpClient(connector)
+	force := d.Get("force").(string)
+	ipAddress := d.Get("ip_address").(string)
+	ipv6Address := d.Get("ipv6_address").(string)
+	_, err := client.Setvirtualip(&force, &ipv6Address, &ipAddress)
+	if err != nil {
+		return fmt.Errorf("Failed to set cluster virtual ip: %s", err)
+	}
+	d.SetId(id)
```
Comment (annakhm) [inline nsxt/resource_nsxt_cluster_virtual_ip.go:98]: Not needed in update flow

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 1 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:36]: In the client function, the type of parameter force is *string is  https://github.com/vmware/vsphere-automation-sdk-go/blob/master/services/nsxt-mp/nsx/cluster/ApiVirtualIpClient.go

Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:36]: `Setvirtualip(forceParam *string, ip6AddressParam *string, ipAddressParam *string)`

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 2 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:51]: Thanks for catching, fixed

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 3 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:47]: Yes, so if we don't configure virtual ip of a cluster, its virtual ip property will be ipv4: 0.0.0.0, ipv6: ::

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 4 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:86]: Removed

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 5 above)
Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:98]: removed.

File: website/docs/r/cluster_virtual_ip.html.markdown [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+subcategory: "Fabric"
+layout: "nsxt"
+page_title: "NSXT: nsxt_cluster_virtual_ip"
+description: A resource to configure virtual IP of NSXT cluster.
+---
+
+# nsxt_manager_cluster
+
+This resource provides a method for configuring the virtual IP of NSXT cluster.
+This resource is supported with NSX 4.1.0 onwards.
```
[the remark this answers is not in the document]
Comment (2ez4szliu) [reply website/docs/r/cluster_virtual_ip.html.markdown:11]: Done. Also added it to manager_cluster_resource.

File: nsxt/resource_nsxt_cluster_virtual_ip.go (hunk 1 above)
Comment (annakhm) [reply nsxt/resource_nsxt_cluster_virtual_ip.go:36]: We can still make this `TypeBool` for the user

Comment (2ez4szliu) [reply nsxt/resource_nsxt_cluster_virtual_ip.go]: done

**Review by annakhm — APPROVED** [review]

**Pull request merged**